metadata.go

```diff
@@ -25,6 +25,7 @@ type KeyspaceMetadata struct {
 	// Deprecated: use the MaterializedViews field instead.
 	Views             map[string]*ViewMetadata
 	MaterializedViews map[string]*MaterializedViewMetadata
+	Types             map[string]*TypeMetadata
```

Comment:
`system_schema.types` hold the info for user-defined types. Should we name this new field `UserTypes` and the struct `UserTypeMetadata` to make it obvious?

Reply:
Good suggestion @jorgebay , done this.